buffer/buffer.go

```diff
@@ -71,7 +73,8 @@ func (bf *BufferFile) Read(p []byte) (n int, err error) {
 	n = copy(p, bf.buff[bf.loc:len(bf.buff)])
 	bf.loc += n
 
-	if bf.loc == len(bf.buff) {
+	// if copied files into buffer with not enaugh capacity
+	if bf.loc < len(bf.buff) {
```

Comment:
@panikgaul 100% unit test coverage is nice.

But, it seems that your change is wrong and the code before the change is correct.

Did you intended `if n < len(p) {` ?

Even so, I don't understand how io.EOF relates to the capacity of the destination buffer.

Reply:
Hello,

yes the `if n < len(p) {` was my intention.

When I used this with `==`, I always got EOF, but that is not correct.
If destination buffer is smaller then source buffer, `copy()` gives you the count of copied bytes -> size of destination bugger, and you should get EOF in this case, because the source was not copied completely. That was my understanding and why I switched to `<`.
Now it works as expected.

I hope it was plausible :)

Reply:
A partial copy due to insufficient capacity in the destination buffer (even if nothing is copied to a destination buffer with a capacity of 0 bytes) is not an error.

The [io.EOF](https://pkg.go.dev/io#pkg-variables) only indicates whether the end of the source has been reached.  
It is absolutely not related at all with the destination buffer.

It is correct behavior for the [Read](https://pkg.go.dev/io#Reader) of [io.Reader](https://pkg.go.dev/io#Reader) to return [io.EOF](https://pkg.go.dev/io#pkg-variables) when the source is fully read.